ccf_transaction_processor/transaction_processor/pdo_tp.cpp

```diff
@@ -541,7 +556,17 @@ namespace ccfapp
             }
 
             if (ledger_signer_local == NULL) {
-                return make_error(HTTP_STATUS_BAD_REQUEST, "Cannot find signer locally");
+                auto signer_view = tx.get_view(signer);
+                auto signer_global = signer_view->get_globally_committed("signer");
+                if (signer_global.has_value()){
+                    auto key_pair = signer_global.value();
+                    auto privk_pem = tls::Pem(key_pair["privk"]);
+                    ledger_signer_local = make_key_pair(privk_pem, nullb, false);
+                }
+                else {
+                    return make_error(
+                        HTTP_STATUS_BAD_REQUEST, "Unable to locate ledger authority for signing read rpcs");
+                }
             }
```

Comment:
using the same code multiple times. does it make sense to pull this out into a utility function that could be re-used?

Reply:
thank you for the observation, i would love to do this, just that i did not know how to do it in this case ! Specifically, this reusable function must take a CCF table_view as an input, and i am not really sure how this gets passed from inside the lambda. I will explore this.

Reply:
given how ugly the enclave invocation lambda is in the eservice code, i'm pretty sure it could be done. however, if it really makes things that complex, then the gain from unifying the code would be lost (kind of my feeling about the eservice invocation code). your call on which creates more complexity. 

you might also be able to use a (very big) macro that would do this. but again... not sure it would simplify.